ibis/backends/clickhouse/compiler/core.py

```diff
@@ -113,13 +113,22 @@ def fn(node, _, **kwargs):
     replace_notany_with_min_not = p.NotAny(x, where=y) >> c.Min(c.Not(x), where=y)
     replace_notall_with_max_not = p.NotAll(x, where=y) >> c.Max(c.Not(x), where=y)
 
+    # subtract one from ranking functions to convert from 1-indexed to 0-indexed
+    subtract_one_from_ranking_functions = p.WindowFunction(
+        p.RankBase | p.NTile
```

Comment:
Can there be an intermediate `ops.Alias`? I assume no in usual circumstances.

Reply:
Yes, an intermediate `Alias` can exist here.

This is a problem, but we should handle it at a different level.

Aliases should be allowed as input anywhere, and then removed everywhere except in three places:

- `ops.Selection.selections`
- `ops.Aggregation.metrics`
- `ops.Aggregation.by`

These are the only places where an alias is required.

Another way to think of it is that there are no `Value` operations where an `Alias` makes sense.

For example:

```python
t.x.name("y") + 1
```

would compile to

```sql
SELECT (x AS y) + 1
```

which isn't a valid operation.